hpy/devel/src/runtime/argparse.c

```diff
@@ -337,11 +342,37 @@ parse_item(HPyContext *ctx, HPyTracker *ht, HPy current_arg, int current_arg_tmp
         break;
     }
 
-    default:
+    case 's': {
+        const char **output = va_arg(*vl, const char **);
+        if (!HPyUnicode_Check(ctx, current_arg)) {
+            set_error(ctx, ctx->h_TypeError, err_fmt, "a str is required");
+            return 0;
+        }
+        HPy_ssize_t size;
+        const char *data = HPyUnicode_AsUTF8AndSize(ctx, current_arg, &size);
+        // loop bounded by size is more robust/paranoid than strlen
```

Comment:
Fixed

Reply:
For this the Python implementation would have to be faulty and return wrong `size`. That's why I also changed the error to `SystemError` (was `ValueError`).

Reply:
You meant the `HPyUnicode_AsUTF8AndSize` returns `NULL` situation. Hmm, I am not sure how to construct such string without some hacks via the C API. On the Python side it would not allow me to create a string with invalid bytes sequence.

Reply:
It being not possible would also explain why CPython uses `SystemError` and not `ValueError`

Reply:
ah indeed, I think you are right. Nevermind then